crates/bevy_ecs/src/archetype.rs

```diff
@@ -31,9 +31,14 @@ impl ArchetypeId {
     }
 }
 
+pub enum ComponentStatus {
+    Added,
+    Mutated,
+}
+
```

Comment:
Does this mean that Mutated will overwrite Added? I'm not sure this behavior is desirable here: you may always want to post-process a component after its addition, but this could be missed if it's mutated before then.

Reply:
It will not override. If the inserted component already exist it will only update its changed counter, otherwise it will set both added and changed counters. I should probably rename Mutated to Changed though.

Reply:
Scratch that, Mutated is the correct term since it's disjoint from Added